src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -164,6 +169,40 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id,
     return S_OK;
   }
 
+  if (debug_logging_enabled) {
+    Debug("ModuleLoadFinished: ", module_id, " ", module_info.assembly.name,
+            " AppDomain ", module_info.assembly.app_domain_id, " ",
+            module_info.assembly.app_domain_name);
+  }
+
+  AppDomainID app_domain_id = module_info.assembly.app_domain_id;
+
+  // Identify the AppDomain ID of mscorlib which will be the Shared Domain
+  // because mscorlib is always a domain-neutral assembly
+  if (!mscorlib_module_loaded && module_info.assembly.name == "mscorlib"_W) {
+    mscorlib_module_loaded = true;
+    mscorlib_app_domain_id = app_domain_id;
+    return S_OK;
+  }
+
+  // Identify the AppDomain ID of the managed profiler entrypoint
+  if (module_info.assembly.name == "Datadog.Trace.ClrProfiler.Managed"_W) {
+    managed_profiler_module_loaded = true;
+    managed_profiler_app_domain_id = app_domain_id;
+  }
+
+  // Do not modify the module if it has been loaded into the Shared Domain
+  // and the profiler is not in the Shared Domain
+  if (mscorlib_module_loaded && managed_profiler_module_loaded &&
+      app_domain_id == mscorlib_app_domain_id &&
+      mscorlib_app_domain_id != managed_profiler_app_domain_id) {
```

Comment:
Does this need an `#ifdef`?

Reply:
I added a check on the runtime information provided by the profiling API. Let me know if you have feedback on that!